src/Components/Quiz/Quiz.tsx

```diff
@@ -0,0 +1,27 @@
+import { useEffect } from 'react';
+import { useParams } from 'react-router';
+import { useQuiz } from '../../context/quizContext';
+import { CurrentQuestion } from './CurrentQuestion';
+
+export const Quiz = () => {
+	const { quizState, quizDispatch } = useQuiz();
+	const { quizId } = useParams();
+
+	useEffect(() => {
+		const findCurrentQuiz = quizState.quiz.find((quiz) => {
+			return quiz.id === quizId;
+		});
+		quizDispatch({ type: 'LOAD_CURRENT_QUIZ', payload: findCurrentQuiz });
+	}, []);
```

Comment:
A very nice approach to handle data on refresh!

Reply:
Thank you